api/docs/v2/storefront/index.yaml

```diff
@@ -4486,6 +4577,153 @@ components:
+    MenuAttributes:
+      type: object
+      properties:
+        name:
+          type: string
+          example: "Main Menu"
+        location:
+          type: string
+          example: "header"
+        locale:
+          type: string
+          example: "en"
+    MenuRelationships:
+      type: object
+      properties:
+        menu_items:
+          type: object
+          description: Menu Items associated with this Menu
+          properties:
+            data:
+              type: array
+              items:
+                $ref: '#/components/schemas/MenuToMenuItemRelation'
+    MenuToMenuItemRelation:
+      type: object
+      properties:
+        id:
+          type: string
+          example: 3
+        type:
+          type: string
+          example: menu_item
+    MenuAttributesAndRelationships:
+      type: object
+      properties:
+        id:
+          type: string
+          example: '1'
+        type:
+          type: string
+          default: menu
+        attributes:
+          $ref: '#/components/schemas/MenuAttributes'
+        relationships:
+          $ref: '#/components/schemas/MenuRelationships'
+    MenuItem:
+      type: object
+      properties:
+        id:
+          type: string
+          example: 3
+          description: Menu Item Id
+        type:
+          type: string
+          example: menu_item
+          description: Menu Item
+        attributes:
+          $ref: '#/components/schemas/MenuItemAttributes'
+        included:
+          type: array
+          items:
+            oneOf:
+              - $ref: '#/components/schemas/Icon'
+    MenuItemAttributes:
+      type: object
+      properties:
+        code:
+          type: string
+          example: promo
+        name:
+          type: string
+          example: View T-Shirts
+        subtitle:
+          type: string
+          example: special offers
+        link:
+          type: string
+          example: /t/categories/mens/t-shirts
+        new_window:
+          type: boolean
+          example: false
+        lft:
+          type: integer
+          example: 3
+        rgt:
+          type: integer
+          example: 4
+        depth:
+          type: integer
+          example: 0
+        is_container:
+          type: boolean
+          example: true
+        is_root:
+          type: boolean
+          example: false
+        is_child:
+          type: boolean
+          example: true
+        is_leaf:
+          type: boolean
+          example: true
+    Icon:
+      type: object
+      properties:
+        id:
+          type: string
+          example: '1'
+        type:
+          type: string
+          default: icon
+        attributes:
+          $ref: '#/components/schemas/IconAttributes'
+    IconAttributes:
+      type: object
+      properties:
+        viewable_type:
+          type: string
+          example: Spree::MenuItem
+        viewable_id:
+          type: string
+          example: 22
```

Comment:
@MatthewKennedy I think we don't need these as attribute, relationship handles this well, no need for duplication

Reply:
I'll give the Menus one a good going over, As I was doing the pages I realised I need to go back over the menus again.